enhancements/sig-architecture/105-aws-iam-registration/README.md

```diff
@@ -0,0 +1,464 @@
+     --cluster-name managed-0
+```
+
+
+#### Story 4 - Hub administrator accepts a managed cluster using AWS IRSA authentication strategy
+
+It must be possible for the cluster administrator to specify they wish to accept the cluster registration requests using `aws-irsa` authentication strategy in the `clusteradm accept` command, so that it can create identities on hub IAM for managed cluster. The default authentication strategy will be `csr`:
+
+```
+% clusteradm accept \
+     --registration-auth=aws-irsa \
+     --clusters managed-0
+```
+
+
+#### Story 5 - Managed cluster administrator un-joins a cluster from the hub using AWS IRSA authentication strategy
+
+It must be possible for the cluster administrator to specify they wish to use `aws-irsa` authentication strategy in the `clusteradm unjoin` command to un-register with hub. The default authentication strategy will be `csr`:
+
+```
+% clusteradm unjoin \
+     --registration-auth=aws-irsa \
+     --cluster-name managed-0
+```
+
+#### Story 6 - EKS Hub administrator must accept the registration request using aws-irsa authentication strategy only, using csr authentication strategy will throw an error
+
+Since EKS doesn't support CSR based client auth and, self managed or clusters running on other cloud providers will not support AWS IRSA, one hub cluster will not support both the authentication strategies at the same time. Running `clusteradm accept` command on EKS with --registration-auth=csr will throw an error as there is no csr to approve, provided both `clusteradm init` and `clusteradm join` were run using --registration-auth=aws-irsa on hub and managed cluster respectively. 
+
+
+#### Story 7 - Procedure for managed cluster administrator to follow to refresh the token
+
+The temporary IAM credentials, for the role assumed by pod, obtained by klusterlet-agent from sts will have TTL of 1 hour by default, which will be [renewed](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Credentials.html#:~:text=Occasionally%20credentials%20can%20expire%20in%20the%20middle%20of%20a%20long%2Drunning%20application.%20In%20this%20case%2C%20the%20SDK%20will%20automatically%20attempt%20to%20refresh%20the%20credentials%20from%20the%20storage%20location%20if%20the%20Credentials%20class%20implements%20the%20refresh()%20method.) by AWS SDK automatically. More info can be found [here](https://docs.aws.amazon.com/eks/latest/userguide/pod-configuration.html).
+
+The EKS kubeconfig will use these temporary IAM creds to generate temporary sts token using `aws eks get-token` command present in kubeconfig with [TTL of 15 mins](https://aws.github.io/aws-eks-best-practices/security/docs/iam/#:~:text=The%20token%20has%20a%20time,each%20time%20the%20token%20expires.). This sts token authenticates every request sent to EKS api-server, and is automatically renewed by kubeconfig on expiry. Snippet of user from an EKS kubeconfig:
+```yaml
+users:
+- name: test-user
+  user:
+    exec:
```

Comment:
We want to keep the design for this solution similar to existing csr based auth that uses a hub kubeconfig from a secret called hub-kubeconfig-secret.

Also we are planning to publish and use separate images for registration and work agent on spoke and registration controller on hub if selected `--registration-auth` is `aws-irsa`, that will contain aws cli. This is done to ensure that csr based solution is not unnecessarily bloated with aws cli or sdk. Whether to use aws or non-aws images will depend upon the value of `--registration-auth` and we will have to update clusteradm for this.

Also if my understanding is correct, we will have to make a lot more changes from the exiting csr solution if we decide to use aws sdk approach instead of loading hub kubeconfig file.

Could you please explain what do you mean by tokenfile, if you are talking about the one mentioned [here](https://docs.aws.amazon.com/eks/latest/userguide/pod-configuration.html), that is OIDC token and cannot be used for authentication with EKS, EKS needs sts token which is generated by `aws eks get-token` command.

Please let me know if i am missing anything here and there is a way to avoid using aws-cli without diverging too much from the existing design.

Reply:
What i mean is sts token, and I am thinking you can use AWS go sdk to get the token instead of using the cmd line

Reply:
Perhaps this could be modeled as a side-car that persists the token to an emptyDir and avoid the dep?

Reply:
@qiujian16 are you able to share an article that shows how that can be done using aws go sdk and how the result token can be used with client-go library?

I like @deads2k init container approach as well as that seems to be less disruptive to existing design.

Reply:
I think it could work with init container.